projects/aca-content/src/lib/components/search/search-save/list/ui-list/saved-searches-list.ui-component.ts

```diff
@@ -0,0 +1,153 @@
+import { SavedSearch } from '@alfresco/adf-content-services';
+import { Clipboard } from '@angular/cdk/clipboard';
+import { Store } from '@ngrx/store';
+import { SnackbarInfoAction } from '@alfresco/aca-shared/store';
+
+@Component({
+  selector: 'aca-saved-searches-ui-list',
+  standalone: true,
+  imports: [CommonModule, DATATABLE_DIRECTIVES, TEMPLATE_DIRECTIVES, DataTableComponent],
+  templateUrl: './saved-searches-list.ui-component.html',
+  styleUrls: ['./saved-searches-list.ui-component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  host: { class: 'aca-saved-searches-ui-list' }
+})
+export class SavedSearchesListUiComponent extends DataTableSchema implements AfterContentInit, OnDestroy {
+  @Input()
+  savedSearches: SavedSearch[] = [];
+
+  @Output()
+  savedSearchOrderChanged = new EventEmitter<{ previousIndex: number; currentIndex: number }>();
+
+  readonly ShowHeaderMode = ShowHeaderMode;
+
+  private savedSearchesListUiService = inject(SavedSearchesListUiService);
+  private onDestroy$ = new Subject<boolean>();
+  private contextMenuAction$ = new Subject<any>();
+
+  private readonly editSavedSearchOptionKey = 'edit';
+  private readonly deleteSavedSearchOptionKey = 'delete';
+  private readonly copyToClipboardUrlOptionKey = 'copy';
+  private readonly menuOptions = [
+    {
+      icon: 'copy',
+      title: 'APP.BROWSE.SEARCH.SAVE_SEARCH.LIST.COPY_TO_CLIPBOARD',
+      key: this.copyToClipboardUrlOptionKey
+    },
+    {
+      icon: 'edit',
+      title: 'APP.BROWSE.SEARCH.SAVE_SEARCH.EDIT_DIALOG.CONTEXT_OPTION',
+      key: this.editSavedSearchOptionKey
+    },
+    {
+      icon: 'delete',
+      title: 'APP.BROWSE.SEARCH.SAVE_SEARCH.DELETE_DIALOG.CONTEXT_OPTION',
+      key: this.deleteSavedSearchOptionKey
+    }
+  ];
+
+  constructor(protected appConfig: AppConfigService, private readonly clipboard: Clipboard, private readonly store: Store) {
+    super(appConfig, '', savedSearchesListSchema);
+  }
+
+  ngAfterContentInit() {
+    this.createDatatableSchema();
+    this.contextMenuAction$.pipe(takeUntil(this.onDestroy$)).subscribe((action) => this.executeMenuOption(action.key, action.data));
+  }
+
+  ngOnDestroy() {
+    this.onDestroy$.next(true);
+    this.onDestroy$.complete();
+  }
+
+  onShowRowActionsMenu(event: DataCellEvent): void {
+    event.value.actions = this.menuOptions;
+  }
+
+  onSearchOrderChange(event: { previousIndex: number; currentIndex: number }): void {
+    this.savedSearchOrderChanged.next(event);
+  }
+
+  executeMenuOption(optionKey: string, savedSearchData: SavedSearch): void {
+    switch (optionKey) {
+      case this.editSavedSearchOptionKey:
+        this.openEditSavedSearchDialog(savedSearchData);
+        break;
+      case this.deleteSavedSearchOptionKey:
+        this.openDeleteSavedSearchDialog(savedSearchData);
+        break;
+      case this.copyToClipboardUrlOptionKey:
+        this.copyToClipboard(savedSearchData);
```

Comment:
Minor point but can you add break in the last case? While at the moment it won't make any difference, it would ensure that if there are more items added to the context menu in the future, then we won't have to modify existing code (however small the modification)